protoc-gen-swagger/defs.bzl

```diff
@@ -0,0 +1,79 @@
+def _collect_includes(srcs):
+    includes = ["."]
+    for src in srcs:
+        include = src.dirname
+        if include and not include in includes:
+            includes += [include]
+
+    return includes
+
+def _run_proto_gen_swagger(proto_service, transitive_proto_srcs, actions, protoc, protoc_gen_swagger):
+    swagger_file = actions.declare_file(
+        "%s.swagger.json" % proto_service.basename[:-len(".proto")],
+        sibling = proto_service,
+    )
+
+    args = actions.args()
+    args.add("--plugin=%s" % protoc_gen_swagger.path)
+    args.add("--swagger_out=logtostderr=true:%s" % swagger_file.dirname)
+    args.add("-Iexternal/com_google_protobuf/src")
+    args.add("-Iexternal/grpc_ecosystem_grpc_gateway/third_party/googleapis/")
```

Comment:
It may be tricky to get the include directory for the protos from the repository target. Let me know if you have any questions regarding that.